module/Kbuild.in

```diff
@@ -51,6 +51,10 @@ endif
 # Suppress unused-value warnings in sparc64 architecture headers
 ccflags-$(CONFIG_SPARC64) += -Wno-unused-value
 
+# Linux 6.2 compat: Do not permit Linux to insert -funsigned-char into CFLAGS.
+# That flag breaks code that expects the universal interpretation of char as a
+# signed type, and would cause the compiler to miscompile our code.
+KBUILD_CFLAGS := $(filter-out -funsigned-char,$(KBUILD_CFLAGS))
```

Comment:
Dumb question - can we just search-n-replace all `char` to `signed char` in `u8_textprep.c` (or at least the ones that convert to an `int`) and remove the `KBUILD_CFLAGS` change?  Or am I missing something obvious?

Reply:
would any of you like for someone to try out doing just that and let you know how it goes?

Reply:
This comment it out of date. I had thought that clang-tidy was only reporting uses of char, but it reported all 8-bit types with widening. The result is that it reported code in u8_textprep.c that was already explicitly signed or unsigned and did not have a problem.

Initially, I wanted to remove this flag from KBUILD_CFLAGS as a safety precaution, but after I learned more following writing that comment, I learned that it was not necessary, but I still thought removing it was a good idea for consistency. Now that I have considered it even more, I think it is find not to remove this flag and that I had overreacted.

I will repush this branch later removing the commit that adds this KBUILD_CFLAGS change in favor of only having the commit fixing the buffer overflow.